File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        int opened = 0;
        int closed = 0;
        int n = s.length();
        for(int i=0; i<n; i++){
            if (s[i] == '('){
                opened ++;
            }
            else{
                if (opened > 0){
                    opened --;
                }
                else{
                    closed ++;
                }
            }
        }
        return opened + closed;
    }
};

File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        opened, closed = 0, 0
        n = len(s)
        for i in range(n):
            if s[i] == '(':
                opened += 1
            else:
                if opened > 0:
                    opened -= 1
                else:
                    closed += 1
        return opened + closed